test/unit/test_lapack_tile.cpp

```diff
@@ -22,12 +24,55 @@ using namespace dlaf_test;
 using namespace testing;
 
 const std::vector<blas::Uplo> blas_uplos({blas::Uplo::Lower, blas::Uplo::Upper});
+const std::vector<blas::Diag> blas_diags({blas::Diag::NonUnit, blas::Diag::Unit});
 
 template <typename Type>
 class TileOperationsTest : public ::testing::Test {};
 
 TYPED_TEST_SUITE(TileOperationsTest, MatrixElementTypes);
 
+TYPED_TEST(TileOperationsTest, lange) {
```

Comment:
The testing is quite comprehensive but we only provide a thin interface over lapackpp. We essentially end-up testing lapackpp for correctness. Wouldn't it be simpler to instead compare the output from `lapackpp::lange` and `lapackpp:lantr` to our wrappers? A single test that checks parameters are passed properly should suffice. I am a bit concerned that we are adding a lot of code that we have to maintain for simple wrapper functions.

Reply:
On the other hand, if we decide to add other back ends, it might be good having the tests reference only the public API and not implementation details.

Reply:
I would leave the tests as they are, relying on the public API is IMO better. You can mark this as resolved unless you want to add something.

Reply:
It's a good point and I agree with your last statement.

About tests, I'm wondering how comprehensive we want to be. For instance, do we want to test all types for norm type? I think that in theory we should, but at the same time the quantity of code, as you were concerned, may easily explode exponentially.

What about this? @rasolca @teonnik

Reply:
I think that adding some basic tests it is always a good idea:
you check if you interface the library correctly, but also check that no strange problems are around:
e.g. header version that do not match the library...
Moreover as you already mention the interface library can be changed easily.
Of course we do not have to check all corner cases as for our implementations, but I would check that the different versions (different parameters for uplo, op, side, ...) are interfacing correctly.

Reply:
Related to our discussion : https://github.com/eth-cscs/DLA-Future/issues/232 .